modules/host/network.go

```diff
@@ -85,9 +85,18 @@ func (h *Host) initNetworking(address string) (err error) {
 // threadedHandleConn handles an incoming connection to the host, typically an
 // RPC.
 func (h *Host) threadedHandleConn(conn net.Conn) {
-	// We defer the conn.Close so that the conn is still closed if `Add` returns
-	// an error.
-	defer conn.Close()
+	// Close the conn immediately on host.Close.
+	defer func() {
+		// We defer the conn.Close so that the conn is still closed if `Add` returns
+		// an error.
+		defer conn.Close()
+		err := h.tg.Add()
+		if err != nil {
+			return
+		}
+		defer h.tg.Done()
+		<-h.tg.StopChan()
+	}()
```

Comment:
hmm... actually, this doesn't perfectly fit the use case. It works great for `net.Listener`s, but not so much for `net.Conn`s. If you naively register every conn, then the ThreadGroup will wind up storing a reference to every conn you've ever made. Definitely bad.